Add documentation on security features (#3799)
theforeman/foreman-documentation: guides/common/modules/con_security-features.adoc

[id="security-features_{context}"]

> = Security features in {Project}

Reviewer: ```suggestion
= Security hardening in {Project}
```
This seems to be the standard term used in RHEL docs.

Author: It is an established term indeed, but I don't think it would be a good heading here. The module is a conceptual/referential list of the available security features, rather than a specific guide how to set them up them. Also, using a gerund heading for a non-procedural module should be avoided according to our style guides.

Reviewer 2: I'd agree it's not a good heading. Many of these are more compliance than hardening. For example, FIPS only has a certainly list of allowed ciphers. Newer ciphers (which are arguably better) may not be on the list until a new standard is allowed. Effectively FIPS can make a system less secure, but compliant. An example we had in the past: FIPS didn't allow the use of MD5, but RHEL 6 only had MD5 checksums for RPMs. The solution: disable checksums altogether.

IMHO nobody should use FIPS unless you're mandated to.

+
For more information, see {RHELDocsBaseURL}9/html/security_hardening/switching-rhel-to-fips-mode_security-hardening[Switching RHEL to FIPS mode] in _{RHEL}{nbsp}9 Security hardening_.

File Access Policy Daemon (`fapolicyd`)::
By enabling `fapolicyd` on your {ProjectServer}, you can monitor and control file and directory access, providing an additional security layer.
+
This feature helps prevent unauthorized code execution and enhances system integrity. You can enable or disable `fapolicyd` on both {Project} and {SmartProxyServer} at any time, depending on your security requirements.
+
For additional information about `fapolicyd`, see {RHELDocsBaseURL}9/html/security_hardening/assembly_blocking-and-allowing-applications-using-fapolicyd_security-hardening[Blocking and allowing applications by using fapolicyd] in _Security hardening_.
+
For instructions on enabling `fapolicyd`, see {InstallingServerDocURL}installing_satellite_server_in_a_connected_network_environment/installing_server_connected_{project-context}#using-fapolicyd-on-server_{project-context}[Using fapolicyd on Satellite Server

> ] in _{InstallingServerDocTitle}_.

Author: Good catch, will fix.

:_newdoc-version: 2.18.4
:_template-generated: 2025-04-23
:_mod-docs-content-type: CONCEPT

[id="security-features_{context}"]
= Security features in {Project}

When deploying {Project}, consider implementing security features that enhance the overall security of your system.
{Project} supports multiple security mechanisms, including FIPS mode, `fapolicyd`, and SELinux, to provide additional layers of protection.

Federal Information Processing Standards (FIPS) Mode::

> {Project} supports running on FIPS-enabled RHEL hosts to comply with security standards for cryptographic modules. 

Reviewer 3: ```suggestion
{Project} supports running on FIPS-enabled {RHEL} hosts to comply with security standards for cryptographic modules. 
```

However, {Project} itself does not currently support running in FIPS mode.
+
FIPS mode enforces the use of validated cryptographic algorithms and modules.
Enabling FIPS mode ensures that all cryptographic operations within the system adhere to strict security standards.

File Access Policy Daemon (`fapolicyd`)::

> By enabling `fapolicyd` on your {Project} Server, you can monitor and control file and directory access, providing an additional security layer.

Reviewer 3: ```suggestion
By enabling `fapolicyd` on your {ProjectServer}, you can monitor and control file and directory access, providing an additional security layer.
```

+
This feature helps prevent unauthorized code execution and enhances system integrity. You can enable or disable `fapolicyd` on both {Project} and {ProjectServer} at any time, depending on your security requirements.
+

> For additional information about `fapolicyd`, see link:https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html/security_hardening/assembly_blocking-and-allowing-applications-using-fapolicyd_security-hardening[Blocking and allowing applications by using fapolicyd] in _Security hardening_.

Reviewer 3: ```suggestion
For additional information about `fapolicyd`, see link:{RHELDocsBaseURL}9/html/security_hardening/assembly_blocking-and-allowing-applications-using-fapolicyd_security-hardening[Blocking and allowing applications by using fapolicyd] in _Security hardening_.
```

[id="security-features_{context}"]
= Security features in {Project}

When deploying {Project}, consider implementing security features that enhance the overall security of your system.
{Project} supports multiple security mechanisms, including FIPS mode, `fapolicyd`, and SELinux, to provide additional layers of protection.

> Federal Information Processing Standards (FIPS) Mode::

Reviewer 2: I'd like a note that you can't enable FIPS on an already installed `{Project}`. Today we have that in our installation guides, but it's better in the planning guide. FIPS mode is neither a prerequisite nor is enabling FIPS part of the installation procedure. Once you have that, you can drop the FIPS mode parts from our installation guides.

{Project} supports running on FIPS-enabled {RHEL} hosts to comply with security standards for cryptographic modules. 

> However, {Project} itself does not currently support running in FIPS mode.

Reviewer 2: I think we usually say `{Project}` itself is not FIPS certified.
```suggestion
However, {Project} itself is not FIPS certified.
```

+
FIPS mode enforces the use of validated cryptographic algorithms and modules.
Enabling FIPS mode ensures that all cryptographic operations within the system adhere to strict security standards.
+
For more information, see {RHELDocsBaseURL}9/html/security_hardening/switching-rhel-to-fips-mode_security-hardening[Switching RHEL to FIPS mode] in _{RHEL}{nbsp}9 Security hardening_.

File Access Policy Daemon (`fapolicyd`)::
By enabling `fapolicyd` on your {ProjectServer}, you can monitor and control file and directory access, providing an additional security layer.
+
This feature helps prevent unauthorized code execution and enhances system integrity. You can enable or disable `fapolicyd` on both {Project} and {SmartProxyServer} at any time, depending on your security requirements.
+
For additional information about `fapolicyd`, see {RHELDocsBaseURL}9/html/security_hardening/assembly_blocking-and-allowing-applications-using-fapolicyd_security-hardening[Blocking and allowing applications by using fapolicyd] in _Security hardening_.
+

> For instructions on enabling `fapolicyd`, see {InstallingServerDocURL}installing_satellite_server_in_a_connected_network_environment/installing_server_connected_{project-context}#using-fapolicyd-on-server_{project-context}[Using fapolicyd on Satellite Server] in _{InstallingServerDocTitle}_.

Reviewer 2: I'm not sure we need to link there. The original goal was to drop the chapter in the installation guide, though https://issues.redhat.com/browse/SAT-30719 doesn't cover that.

For a user it's sufficient to run `dnf install fapolicyd`, either before or after installation. https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html/security_hardening/assembly_blocking-and-allowing-applications-using-fapolicyd_security-hardening (where you already link to in the paragraph above) does exactly that. In other words: there's nothing in our procedures that adds value. And the group of users who use `fapolicyd` will already know what `fapolicyd` is. Those who don't know it, don't need it.

The conclusion was that we should mention in the planning guide that we support fapolicyd (and will ensure the right policy is installed). Then drop `guides/common/modules/proc_installing-fapolicyd-on-server.adoc` and all references to it.

IMHO that should be part of this PR.

Author: I see, I'll work on including that in the next draft, then.

Reviewer 2: I think that reads well to me.

Security-Enhanced Linux (SELinux)::
{Project} supports SELinux, a mandatory access control system that restricts system access and reduces the risk of security vulnerabilities.
SELinux enforces policies that define which processes can access specific system resources, providing protection against unauthorized access and exploitation.
+
Running {Project} with SELinux in enforcing mode is recommended to maintain a high-security environment.
It ensures that access control policies are strictly enforced, minimizing the risk of privilege escalation or unauthorized actions.
+

> Using SELinux is required for {Project} Server to be installed on your environment. For additional information, see {InstallingServerDocURL}preparing_your_environment_for_installation_{project-context}#system-requirements_{project-context}[System requirements] in _{InstallingServerDocTitle}_.

Reviewer 2: Like SELinux and fapolicyd, the idea was to move it from the installation guide to the planning guide.

Also, technically it's not required but for Satellite it must be enabled to consider the system supported. It may be in permissive mode, though enforcing mode is recommended. For upstream I'd follow those recommendations.

Here the argument to move it isn't as strong because you can consider it a prerequisite.

> :_newdoc-version: 2.18.4
> :_template-generated: 2025-04-23
> :_mod-docs-content-type: CONCEPT
> 

Reviewer: ```suggestion
```
Pls, remove.
FYI, there's the `-D` option in newdoc to avoid this metadata header.

+
For more information, see {RHELDocsBaseURL}9/html/security_hardening/switching-rhel-to-fips-mode_security-hardening[Switching RHEL to FIPS mode] in _{RHEL}{nbsp}9 Security hardening_.

File Access Policy Daemon (`fapolicyd`)::
By enabling `fapolicyd` on your {Project} Server, you can monitor and control file and directory access, providing an additional security layer.
+
This feature helps prevent unauthorized code execution and enhances system integrity. You can enable or disable `fapolicyd` on both {Project} and {SmartProxyServer} at any time, depending on your security requirements.
+
For additional information about `fapolicyd`, see {RHELDocsBaseURL}9/html/security_hardening/assembly_blocking-and-allowing-applications-using-fapolicyd_security-hardening[Blocking and allowing applications by using fapolicyd] in _Security hardening_.
+
For instructions on enabling `fapolicyd`, see {InstallingServerDocURL}installing_satellite_server_in_a_connected_network_environment/installing_server_connected_{project-context}#using-fapolicyd-on-server_{project-context}[Using fapolicyd on Satellite Server
] in _{InstallingServerDocTitle}_.

> 
> https://docs.redhat.com/en/documentation/red_hat_satellite/6.16/html/installing_satellite_server_in_a_connected_network_environment/installing_server_connected_satellite#using-fapolicyd-on-server_satellite

Reviewer 2: Debug left over?
```suggestion
```

Security-Enhanced Linux (SELinux)::
{Project} supports SELinux, a mandatory access control system that restricts system access and reduces the risk of security vulnerabilities.
SELinux enforces policies that define which processes can access specific system resources, providing protection against unauthorized access and exploitation.
+
Running {Project} with SELinux in enforcing mode is recommended to maintain a high-security environment.
It ensures that access control policies are strictly enforced, minimizing the risk of privilege escalation or unauthorized actions.
+

> Using SELinux in enforcing mode is required for {Project} Server to be installed on your environment. For additional information, see {InstallingServerDocURL}preparing_your_environment_for_installation_{project-context}#system-requirements_{project-context} in _{InstallingServerDocTitle}_.

Reviewer 2: This contradicts each other: first it's recommended but then it's required. In the installation guide we say:
> SELinux must be enabled, either in enforcing or permissive mode.

In other words, it's currently recommended (and IMHO that's still correct).

[id="security-features"]
= Security features in {Project}

When deploying {Project}, consider implementing security features that enhance the overall security of your system.

> {Project} supports multiple security mechanisms to provide additional layers of protection, which you can enable during or after the installation of {Project}, if your use case requires them. 

Reviewer: Should we mention that these security considerations are applicable to both ProjectServer and SmartProxyServers?

Reviewer: @jherrman Actually, can you address this? I missed it during review.

Author: @Lennonka , already did: https://github.com/theforeman/foreman-documentation/pull/3799/files#diff-5a886518aa8c58822600953b3d423d55082b31748c85b9f4810099e61c3de04aR9

Is this what you had in mind?

Cheers,
J.

Reviewer: Oh, yes, that. Thanks!

{Project} supports running on FIPS-enabled {RHEL} hosts to comply with security standards for cryptographic modules. 
However, {Project} itself is not FIPS-certified. In addition, you cannot enable FIPS mode after the installation of {Project}.
+
FIPS mode enforces the use of validated cryptographic algorithms and modules.
Enabling FIPS mode ensures that all cryptographic operations within the system adhere to strict security standards.
+
For more information, see {RHELDocsBaseURL}9/html/security_hardening/switching-rhel-to-fips-mode_security-hardening[Switching RHEL to FIPS mode] in _{RHEL}{nbsp}9 Security hardening_.
+
[NOTE]
====
{Project} supports DEFAULT and FIPS crypto-policies.
The FUTURE crypto-policy is not supported for {Project} and {SmartProxy} installations.
The FUTURE policy is a stricter forward-looking security level intended for testing a possible future policy.
For more information, see {RHELDocsBaseURL}9/html/security_hardening/using-the-system-wide-cryptographic-policies_security-hardening[Using system-wide cryptographic policies] in _{RHEL}{nbsp}9 Security hardening_.
====

// we do not package "fapolicy" for Foreman on Debian/Ubuntu
ifndef::foreman-deb[]

> File Access Policy Daemon (`fapolicyd`)::

Reviewer: Oh, I see. Have I mistaken it with SELinux perhaps?

> File Access Policy Daemon (`fapolicyd`)::

Author: Yep, that's already in :-)
